.pre-commit-config.yaml

```diff
@@ -1,9 +1,19 @@
 repos:
+  # create a custom terraform validate for the examples, as launching the validation in the root dir won't work and give
+  # "Error: Provider configuration not present" error
+  # https://github.com/hashicorp/terraform/issues/21416
+  - repo: local
+    hooks:
+        - id: terraform_validate
+          name: Terraform validate
+          entry: .pre-commit-terraform-validate-examples.sh
+          pass_filenames: false
+          language: script
+
```

Comment:
there are some inconsitencies as how terraform validate works in the root dir, vs. examples dir.
within examples, provider aliases work nice, but within the root dir, `terraform validate` gets the error mentioned in the comment "Error: Provider configuration not present" https://github.com/hashicorp/terraform/issues/21416

as I didn't want to loose the validation both in the pre-hook and ci , i added a custom script that iterates through all the examples and validates them.